### ISSUE TITLE
Default Kane's delivery charge to $39.99 when a customer purchases bedding only (for 'K' shipping). Should be like recliners and chairs.

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#860]
_Imported from trac ticket 860.
Created by **kbower**
Opened in trac: Apr 19, 2011, 11:44:35 AM
Last modified in trac: Jan 11, 2012, 3:46:10 PM_
***

**Comment by kbower** [comment]
_**kbower** trac comment on Apr 19, 2011, 12:08:53 PM:_

Wayne,
Only minor '500' or all the ones currently defined including ('500','881','570','513'), etc?

**Comment by kbower** [comment]
_**kbower** trac comment on Jan 9, 2012, 9:39:33 AM:_

Kent or Rob,

I would like the 'K' delivery zone to default to $39.99 for any order with only bedding inventory items, i.e. items where the ITM.MNR_CD in ('500', '881', '513', '523', '530', '538', '551', '561', '565', '570').

Thanks,

*Wayne Liburd*

**Comment by kbower** [comment]
_**kbower** trac comment on Jan 9, 2012, 9:39:56 AM:_

Ticket [#1323](1323) has been marked as a duplicate of this ticket.

**Comment by kbower** [comment]
_**kbower** trac comment on Jan 11, 2012, 10:58:00 AM:_

But in the case of special delivery, then we don't add the $39.99, I assume, but rather set the charge as only the upcharge.  Is that accurate?

**Comment by kbower** [comment]
_**kbower** trac comment on Jan 11, 2012, 11:23:35 AM:_

Kent,

That is correct.

Thanks,

*Wayne Liburd*

**Comment by kbower** [comment]
_**mpisarski** trac comment on Jan 11, 2012, 11:30:24 AM:_

(Changeset [[2138]](../commit/911f957a52c9b0dd7d5a2683a6c91232148f65ce))
 * fixes #860 - Default Kane's delivery charge to $39.99 when a customer purchases bedding only (for 'K' shipping). Should be like recliners and chairs. Also, address special delivery and upcharges for zips.

**Comment by kbower** [comment]
_**mpisarski** trac status change on Jan 11, 2012, 11:30:24 AM:_ **closed**

**Comment by kbower** [comment]
_**mpisarski** trac comment on Jan 11, 2012, 12:45:19 PM:_

Wayne,

A couple more questions:

   - the minor codes you specified for bedding: ITM.MNR_CD in ('500',
   '881', '513', '523', '530', '538', '551', '561', '565', '570')
      - do these apply just for the discounted delivery charge? or do the
      apply for all bedding related rules (eg. discount exclusion rule for
      bedding around thanksgiving.)
   - If the order has only recliners and bedding on it should the
   discounted delivery charge still apply? Or should it only apply if it is
   fully either a bedding order or a recliner order?

Thanks,

Michael

**Comment by kbower** [comment]
_**kbower** trac comment on Jan 11, 2012, 1:20:04 PM:_

Michael,

1. They apply to the discounted delivery charge and

2. Only recliners or bedding not mixed.

Wayne

**Comment by kbower** [comment]
_**kbower** trac comment on Jan 11, 2012, 1:24:12 PM:_

On 1/11/2012 1:09 PM, Wayne Liburd wrote:

Wayne, that part was clear... are you saying that for thanksgiving PRE type of discounts (where you want to exclude bedding), the definition of bedding should remain minor code /strictly/ in ('500','881')?

**Comment by kbower** [comment]
_**kbower** trac comment on Jan 11, 2012, 1:36:36 PM:_

That is correct.

Thanks,

Wayne

**Comment by kbower** [comment]
_**mpisarski** trac comment on Jan 11, 2012, 1:36:39 PM:_

(Changeset [[2139]](../commit/f7348bee92ac102cddb1a9014cc839a3e4399a5e))
 * re #860 - rule tweaks

**Comment by kbower** [comment]
_**kbower** trac comment on Jan 11, 2012, 2:52:42 PM:_

see [#1323](1323).  We need to include removing and adding lines as dependency... this is a bit more complicated.

**Comment by kbower** [comment]
_**kbower** trac status change on Jan 11, 2012, 2:52:42 PM:_ **reopened**

**Comment by kbower** [comment]
_**kbower** trac comment on Jan 11, 2012, 3:20:36 PM:_

Wayne,
    The recliner shipping allows for the exceptions of products '393000004', '086900125', and '086900101'.

Should bedding shipping be $39.99 even if one or more of these products ('393000004', '086900125', '086900101') are on the order?

Kent

**Comment by kbower** [comment]
_**kbower** trac comment on Jan 11, 2012, 3:32:50 PM:_

(Changeset [[2140]](../commit/64f002dc6fbd46ef1db8b0a72f6697d296f02b2b))
 * make Kane's bedding shipping recalculate when order no longer qualifies for either recliner or bedding discounted shipping (closes #860)

**Comment by kbower** [comment]
_**kbower** trac status change on Jan 11, 2012, 3:32:50 PM:_ **closed**

**Comment by kbower** [comment]
_**kbower** trac comment on Jan 11, 2012, 3:46:10 PM:_

Kent,

I included most of those with the additional minors I sent you. The skus you listed are leather protection and swivel bases which are particular to recliners but I included similar related products like mattress pads, pillows, sheets, frames etc. in the list I gave you for bedding.  It is possible something additional may pop up but I think I covered most if not all of the bases.

Thanks,

*Wayne Liburd*
